File: two_pointers/167_two_sum_sorted.py
def two_sum_sorted(nums: list[int], target: int) -> list[int]:
    l = 0
    r = len(nums) - 1
    while True:
        two_sum = nums[l] + nums[r]
        if two_sum == target:
            return [l + 1, r + 1]

        if two_sum > target:
            r -= 1
        else:
            l += 1

File: two_pointers/167_two_sum_sorted.go
func twoSum(numbers []int, target int) []int {
	l := 0
	r := len(numbers) - 1

	for l < r {
		twoSum := numbers[l] + numbers[r]
		if twoSum == target {
			return []int{l + 1, r + 1}
		}

		if twoSum < target {
			l++
		} else {
			r--
		}

	}
	return []int{}
}
